apps/app_utf8.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2018 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include "apps.h"
+#include <openssl/asn1.h>        /* For UTF8_putc */
+
+/* Decide on an implementation */
+#if defined(_WIN32)
+/* Windows has its own stuff */
+# define IMPLEMENT_WIN32
+#elif defined(__linux) || defined(__VMS)
```

Comment:
`__GLIBC__>=2` was suggested as more appropriate way to detect the capability on Linux. Minor challenge is that it's not pre-defined by compiler, so one has to ensure that one of headers indirectly defines it. string.h does. One can wonder if one has to check even `__GLIBC_MINOR__`. At least I can confirm that everything works as intended with `__GLIBC_MINOR__` being 2...

Reply:
Ah, thanks. forgot that.  Will do.

Reply:
Is that also to suggest that this would be workable with glibc on other platforms?

Reply:
I would expect that to be the case; it's a function of the C library, not the underlying kernel.

Of course for glibc this is an utterly pointless special case anyway, since glibc itself *provides* iconv; it's not a separate _(but still fairly much ubiquitous)_ library as it is on Solaris/*BSD.

```
$ nm -D /usr/lib64/libc.so.6 | grep iconv
0000000000023830 T iconv
00000000000239f0 T iconv_close
0000000000023550 T iconv_open
```

Reply:
Sure. And I fail to see why it would be inappropriate.

Reply:
> it's not a separate (but still fairly much ubiquitous) library as it is on Solaris/*BSD.

Right, "something almost, but not quite entirely unlike [tea]." When I said that "Anyone suggesting otherwise should be required to provide a full example ..." is an unreasonable suggestion I suspected exactly this kind of thing. I mean there is contradiction. Assertion sounds quite categorical all-or-nothing, but then it boils down to "fairly"... Come on!